### DISCUSSION TITLE
[Support] @midwayjs/typegoose组件的健康检查报错

### DISCUSSION THREAD

**Discussion by ghostoy** [midwayjs/midway#4056] [Q&A] [answered]
### Describe the problem（描述问题）

健康检查代码
```typescript
import { Controller, Get, Inject, MidwayHealthService } from '@midwayjs/core';

@Controller('/_health')
export class HealthController {
  @Inject()
  healthService: MidwayHealthService;

  @Get('', { summary: '健康检查' })
  async status() {
    return this.healthService.getStatus();
  }
}
```

输出结果
```javascript
{
    "status": false,
    "namespace": "mongoose",
    "reason": "Cannot read properties of undefined (reading 'default')",
    "results": [
        {
            "namespace": "redis",
            "status": true,
            "reason": ""
        },
        {
            "status": false,
            "namespace": "mongoose",
            "reason": "Cannot read properties of undefined (reading 'default')"
        }
    ]
}
```

### Midway Versions（Midway 版本）

>> Start to check your midway component version...

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(not installed)
✓ @midwayjs/axios(3.16.8)
✓ @midwayjs/bootstrap(3.16.6)
✓ @midwayjs/bull(3.16.8)
✓ @midwayjs/bull-board(3.16.8)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.16.2)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(not installed)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(not installed)
✓ @midwayjs/info(not installed)
✓ @midwayjs/jwt(3.16.8)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.16.5)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/mqtt(not installed)
✓ @midwayjs/oss(3.16.5)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(3.16.5)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(not installed)
✓ @midwayjs/socketio(3.16.5)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/tenant(not installed)
✓ @midwayjs/typegoose(3.16.8)
✓ @midwayjs/typeorm(not installed)
✓ @midwayjs/upload(not installed)
✓ @midwayjs/validate(3.16.8)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(3.16.8)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(not installed)
✓ @midwayjs/ws(not installed)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by ghostoy** [comment]
报错定位到 https://github.com/midwayjs/midway/blob/5672afcfb312652455c96007f3cfa667fa17d7bc/packages/core/src/common/dataSourceManager.ts#L229-L231

这里的`this.dataSourcePriority`没有初始化，是`undefined`

**Reply by czy88840616** [reply]
MidwayPriorityManager 属于  midway 中自带的注入类，这个不会没有的。

**Reply by czy88840616** [reply]
如果没有。。那是你的 core 版本太低了不对了吧

**Reply by ghostoy** [reply]
不是MidwayPriorityManager没有，是它的dataSourcePriority字段没有初始化。

**Comment by ghostoy** [comment]
而且mongoose模块的健康检查逻辑有点奇怪：
https://github.com/midwayjs/midway/blob/5672afcfb312652455c96007f3cfa667fa17d7bc/packages/mongoose/src/configuration.ts#L47-L68

为什么已经建立连接的数据源必须是低优先级才能认为是ready呢？

**Reply by czy88840616** [reply]
是必须连上，且非低优先级才报健康检查失败，低优先级相当于弱依赖，允许不健康，不会在健康检查透出

**Reply by ghostoy** [reply]
连上，非低优先级，为什么认为健康检查失败呢？不应该是没有连上才报失败吗？

**Reply by czy88840616** [reply]
哈哈，说的对，逻辑反了

**Comment by czy88840616** [comment, marked as answer]
fix in https://github.com/midwayjs/midway/pull/4061
